doc/quick-start.rst

```diff
@@ -6,10 +6,148 @@ This document gives simple usage examples of Dune. You can also look at
+(replacing ``project_name`` with the name of your project):
+
+.. code:: shell
+
+    dune init proj project_name
+
+This will create a directory for the project that includes the following contents:
+
+.. code::
+
+    project_name/
+    ├── dune-project
+    ├── test
+    │   ├── dune
+    │   └── project_name.ml
+    ├── lib
+    │   └── dune
+    ├── bin
+    │   ├── dune
+    │   └── main.ml
+    └── project_name.opam
+
+Now, enter your project's directory:
+
+.. code:: shell
+
+   cd project_name
+
+Then, you can build your project with:
+
+.. code:: shell
+
+   dune build
+
+You can run your tests with:
+
+.. code:: shell
+
+   dune test
+
+You can run your program with:
+
+.. code:: shell
+
+   dune exec project_name
+
+The following itemization of the generated content isn't necessary to review at
+this point. But whenever you are ready, it will provide jump-off points from
+which you can dive deeper into Dune's capabilities:
+
+* The ``dune-project`` file specifies metadata about the project, including its
+  name, packaging data (including dependencies), and information about the
+  authors and maintainers.  You should open this in your editor to fill in the
+  placeholder values. See :ref:`dune-project` for details.
+* The ``test`` directory contains a skeleton for your project's tests. Add to
+  the tests by editing ``test/project_name.ml``. See :ref:`writing-tests` for
+  details on testing.
+* The ``lib`` directory will hold the library you write to provide the core
+  functionality of your executable. Add modules to your library by creating new
+  ``.ml`` files in this directory. See :ref:`library` for details on specifying
+  libraries manually.
+* The ``bin`` directory holds a skeleton for the executable program. Within the
+  modules in this directory, you can access the modules in your ``lib`` under
+  the namespace ``Project_name.Mod``, where ``Project_name`` is replaced with
+  the name of your project and ``Mod`` corresponds to the name of the file in
+  the ``lib`` directory. You can run the executable with ``dune exec
+  project_name``.  See :ref:`hello-world-program` for an example of specifying
+  an executable manually and :ref:`executable` for details.
+* The ``project_name.opam`` file will be freshly generated from the
+  ``dune-project`` file whenever you build your project. You shouldn't need to
+  worry about this, but you can see :ref:`opam-generation` for details.
+* The ``dune`` files in each directory specify the component to be built with
+  the files in that directory. For details on dune files, see :ref:`dune-files`.
+
+Initializing a library
+--------------------------
+
+To initialize a project for an ocaml library, run the following (replacing
+``project_name`` with the name of your project):
+
+.. code:: shell
+
+    dune init proj --kind=lib project_name
+
+This will create a directory for the project that includes the following contents:
```

Comment:
This creates a project directory that includes the following contents:
...or...
This creates a directory for the project that includes the following contents:
(or it's fine the way it is, if you prefer)

Reply:
I find think those are both better, and the first  one preferable. Thank!